### DISCUSSION TITLE
Why Node.js v18?

### DISCUSSION THREAD

**Discussion by Suyun114** [VSCodium/vscodium#1946] [Q&A]
[microsoft/vscode](https://github.com/microsoft/vscode) requires Node.js v20, but VSCodium still requires v18. And it seems that building VSCodium with Node.js v20 will be crashed. Are there any technical reasons?

**Comment by Jeong-ilKajokaya** [comment]
Hi! Here’s an explanation about Node.js versions in VSCodium vs upstream VSCode:

---

### 1. Node.js Version Differences

* **VSCode** (upstream) now requires Node.js v20 for building and running because newer features and dependencies rely on APIs and ECMAScript features in Node 20.
* **VSCodium** is a community build that mirrors VSCode but sometimes lags in updating Node.js requirements. Currently, it still targets Node.js v18.

### 2. Why VSCodium crashes with Node.js v20

* VSCodium’s build scripts and some patches (removing Microsoft branding, telemetry, etc.) were developed for Node v18.
* Using Node v20 can break scripts due to:

  * Changes in Node APIs
  * Deprecated modules
  * Version-specific dependencies in the `yarn`/`npm` lock files

### 3. Technical considerations

* VSCodium is a **fork with modifications**, so it cannot always adopt the latest upstream build requirements immediately.
* Updating Node.js for VSCodium would require testing all build scripts, dependencies, and patches to ensure compatibility.

### 4. Recommendation

* Stick to **Node.js v18** when building VSCodium for now.
* If you want to try Node.js v20, you would need to patch the build scripts and dependencies — this is non-trivial and not officially supported.

**Summary:**

* VSCode requires Node.js v20; VSCodium is still on v18.
* Crashes happen because VSCodium build scripts aren’t yet compatible with Node 20.
* Stick to Node v18 to build VSCodium reliably.
